src/R/Components/Impl/ConnectionManager/Implementation/ConnectionManager.cs

```diff
@@ -50,9 +50,11 @@ internal class ConnectionManager : IConnectionManager {
 
             _disposableBag = DisposableBag.Create<ConnectionManager>()
                 .Add(_statusBarViewModel)
-                .Add(() => _sessionProvider.BrokerStateChanged -= BrokerStateChanged);
+                .Add(() => interactiveWorkflow.RSession.Connected -= RSession_Connected)
+                .Add(() => interactiveWorkflow.RSession.Disconnected -= RSession_Disconnected);
 
-            _sessionProvider.BrokerStateChanged += BrokerStateChanged;
+            interactiveWorkflow.RSession.Connected += RSession_Connected;
```

Comment:
Sure!

---
In reply to: [80280797](https://github.com/Microsoft/RTVS/pull/2393#discussion_r80280797) [](ancestors = 80280797)